# Практическое задание к 5 семинару (Тестирование гипотез)
## Задание 1
```Когда используется критерий Стьюдента, а когда Z –критерий?```

**Ответ: z-критерий используем когда известна \sigma генеральной совокупности. Если \sigma генеральной совокупности неизвестна то используем критерий Стьюдента**

## Задача 2
```Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу,
если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм,
а дисперсия известна и равна 4 кв. мм.
```

$X_{ср} = 17$

$\alpha = 0.05$

$n = 100$

$\mu = 17.5$

$S = 4$

Используем z-критерий:

$z = \frac {X_{ср} - \mu}{\sqrt{S} / \sqrt{n}} = \frac {17 - 17.5}{\frac{\sqrt{4}}{\sqrt{100}}} = 2.5$


$по\ таблице\ для\ \alpha = 0.05\,,   z_{стат} = 1,645$

$z_{таб} = 1,645 < z = 2.5$

**Ответ: Принимаем альтернативную гипотезу так как расчетные значения больше табличных**

## Задание 3
```
 Проведите тест гипотезы.
 Продавец утверждает, что средний вес пачки печенья составляет 200 г.
 Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
 202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
 Известно, что их веса распределены нормально.
 Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)
```

$X_{ср} = 200$

$n = 10$

$V = n - 1 = 10 - 1 = 9$

$\mu = 198.5$

$P(x) = 1 - \alpha = 0.99 --> \alpha = 0.01 $

$Найдём\ \sigma\ для\ выборки:$

In [1]:
import numpy as np
x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
np.std(x, ddof = 1)

4.453463071962462

Используем t-критерий Стьюдента:

$t = \frac {X_{ср} - \mu}{\sqrt{S} * \sqrt{n}} = \frac {200 - 198.5}{\frac{4.45}{\sqrt{10}}} = 1.066$

$Для\ \frac{\alpha}{2}\ и\ степенью\ свободы\ v = 9,\; t_{стат} = 3.25$

$t_{стат} = 3.25 > t = 1.066$

**Ответ: Принимаем нулевую гипотезу так как критическое значение больше расчетных значений, следовательно существенных отличий нет**

### Задание 4
```
Есть ли статистически значимые различия в росте дочерей?
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160
```

In [5]:
from scipy import stats

x = np.array([172, 177, 158, 170, 178, 175, 164, 160, 169])
y = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])

stats.ttest_rel(x, y)

Ttest_relResult(statistic=0.559522990335608, pvalue=0.5911212354055175)

**Ответ: p-value > $\alpha$ --> статистических значимых различий в росте нет. При $\alpha$ = 0,05**